tests/pcap/src/handshake_message.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::*;
+use rtshark::Packet;
+use rtshark::RTSharkBuilder;
+use std::collections::HashSet;
+
+use crate::capture::*;
+
+#[derive(Debug, Clone, Default)]
+pub struct Message {
+    /// TLS handshake message type
+    pub message_type: u8,
+    /// Packet containing handshake message metadata
+    pub packet: Packet,
+    /// Ordered TCP payloads as hex encoded strings
+    pub payloads: Vec<String>,
+    /// Id / number of the TCP frame containing handshake message metadata
+    pub frame_num: String,
+}
+
+impl Message {
+    const FRAGMENT: &'static str = "tls.handshake.fragment";
+    const FRAGMENTS_COUNT: &'static str = "tls.handshake.fragment.count";
+
+    /// Returns the complete handshake message in bytes.
+    pub fn to_bytes(&self) -> Result<Vec<u8>> {
+        let mut bytes = Vec::new();
+        for payload in &self.payloads {
+            // Trim off the 5-byte record headers.
+            bytes.extend(&hex::decode(payload)?[5..]);
+        }
+        Ok(bytes)
+    }
+
+    fn from_packet(packet: Packet, frame_num: String, tcp_payloads: &Vec<Packet>) -> Result<Self> {
```

Comment:
What are the different responsibilities of `packet` vs `tcp_payloads`. Presumably `tcp_paylods` is disjoint with the main `packet`? If so, why?

Reply:
I added a bunch more comments. Hopefully that helps.